default/scripting/ship_parts/Colony/CO_COLONY_POD.focs.txt

```diff
@@ -14,6 +14,19 @@ Part
 // Population consumption is commented out until the discrepancy between outpost
 // population consumption and this consumption is regularized.
 //  consumption = Population consumption = 1 condition = Source
+    effectsgroups = [
+        EffectsGroup
+            scope = Source
+            activation = Source
+            effects = [
+                GenerateSitrepMessage
+                    message = "SITREP_TUTORIAL_COLONY_POD"
+                    label = "TUTORIAL_HINTS"
+                    icon = "icons/sitrep/beginner_hint.png"
+                    maxshow = 1
```

Comment:
can probably replace this new parameter with a test based on the turn a tech was researched or turn of the game

Reply:
Unfortunately, i don't see how that could be done because CON_OUTPOST could be researched in an arbitrary turn of the game, and then there could be an arbitrary delay before the first ship with a CO_COL_OUTPOST is built.  So to me, whether this is the first turn where a CO_COL_OUTPOST exists seems neither related to which turn of the game it is nor to when CON_OUTPOST was researched.  Besides, it could happen that the first two CO_COL_OUTPOST are built on the same turn, and then we don't want two copies of this sitrep shown.  So i fear the new "maxshow" parameter is needed.

Reply:
A possible workaround (if not Python) might involve adding tracking of the turn on which an empire produces ship parts for the first time, and to base the activation condition on the oldest age of a ship the empire owns, which has that part, being equal to the same turn as that part was first produced.

Reply:
As you recommended, i moved this stuff to Python in commit 76b51873, getting rid of the ugly maxshow parameter.

Note that on the Python side, sitrep.typeString was so far unused.  Trying to use it failed for me:

    Uncaught exception: Traceback (most recent call last):
    File ".../default/python/turn_events/turn_events.py", 
        line 73, in execute_turn_events tutotial_colony_pod()
    File "../default/python/turn_events/turn_events.py", 
        line 282, in tutotial_colony_pod
    print "typeString =", sitrep.typeString
    ArgumentError: Python argument types in None.None(sitrep)
    did not match C++ signature: None(VarText {lvalue})

Only after writing the trivial wrapper function GetTypeString() 
would python access to the C++ method SitRepEntry.GetTemplateString()
start to work for me.  That's included in the patch now.

So i think this conversation can now also be resolved.